src/App.tsx

```diff
@@ -15,36 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods =
+  | 'Dumplings'
+  | 'Carrot'
+  | 'Eggs'
+  | 'Ice cream'
+  | 'Apple'
+  | 'Bread'
+  | 'Fish'
+  | 'Honey'
+  | 'Jam'
+  | 'Garlic';
+
+enum SortType {
+  None = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getPreparedGoods(
+  goods: Goods[],
+  field: SortType,
+  isReversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (field) {
+    preparedGoods.sort((good1, good2) => {
+      switch (field) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [field, setField] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, field, isReversed);
+  const Reset = () => {
```

Comment:
```suggestion
  const reset = () => {
```

Reply:
This function will be created on every render, optimize it via a hook `useCallback`